lib/puppet/indirector/request.rb

```diff
@@ -42,7 +42,8 @@ def environment=(env)
   end
 
   def escaped_key
-    URI.escape(key)
+    # TODO: these keys are generally fragments so Puppet::Util.uri_escape isn't really appropriate
+    Puppet::Util.uri_escape(key)
   end
```

Comment:
Need to determine if this is actually a problem or not... it doesn't make tests fail.